### ISSUE TITLE
Transpile with J2Objc

### ISSUE THREAD

**Issue by confile** [goodow/realtime-json#3]
Do I have to include another lib with translating my Java code to ObjectiveC with you realtime-JSON lib?

**Comment by larrytin** [comment]
Include realtime-json-0.5.5-SNAPSHOT.jar at your class path when
translating Java sources。

BTW，you can find the translated objective-c files at
https://github.com/goodow/GDJson which are generated using
https://github.com/goodow/realtime-json/blob/master/Makefile.

**Comment by confile** [comment]
My question is do I just have to transpile the Java code and I am done or do I have to transpile and include this: https://github.com/goodow/GDJson ?

**Comment by drmichaelgorski** [comment]
Let me rephrase the question. I will use you JSON lib in a Java project. The java project should be translated with (https://github.com/google/j2objc). 

My question is do I just have to transpile the Java code and I am done or do I have to transpile and include this: https://github.com/goodow/GDJson ?

**Comment by larrytin** [comment]
Translate only your Java sources，do NOT translate realtime-json.

**Comment by drmichaelgorski** [comment]
Do I have to include this:https://github.com/goodow/GDJson in my ObjectiveC project?

**Comment by larrytin** [comment]
When translating your java sources to objective-c sources, using these
j2objc options: j2objc --mapping method-mappings.properties
https://github.com/goodow/realtime/blob/master/resources/j2objc/method-mappings.properties
 --prefixes package-prefixes.properties
https://github.com/goodow/realtime/blob/master/resources/j2objc/package-prefixes.properties
.
When using your translated objective-c sources, include GDJson in your
objective-c project.

**Comment by drmichaelgorski** [comment]
Okay I get it. Thank you. 

Can I just include the Classes folder: https://github.com/goodow/GDJson/tree/master/Classes as linked group in my main ObjectiveC project?

**Comment by drmichaelgorski** [comment]
Do I have to modify the translated ObjectiveC classes somehow to include the GDJSON lib? Do I have to add some Imports?

**Comment by larrytin** [comment]
Cocoapods is the recommended way to add GDJson to your oc project.

If you don't use Cocoapods, add these sources to your project: '
Classes/common', 'Classes/generated/*_/_.{h,m}'

Include GDJson wherever you need it with #import "GDJson.h".

**Comment by drmichaelgorski** [comment]
The `#import "GDJson.h"` I have to do myself they are not part of the generated classes right? So I have to add this import in every generated class .h where I use your lib. 
1. Is this right? 
2. Any thing else I have to import?

**Comment by larrytin** [comment]
Don't modify GDJson/Classes/generated/*\* manually, they are generated by
https://github.com/goodow/realtime-json/blob/master/Makefile

Normally you don't need GDJson.h

**Comment by drmichaelgorski** [comment]
This is not what I meat. I mean to I have to modify my generated classed which uses your JSON lib and add #import "GDJson.h" to them? If so, do I have to add anything else?

**Comment by larrytin** [comment]
You don't need to modify your generated files also. J2objc translator will
include every specific header file which you depend on for you.

**Comment by drmichaelgorski** [comment]
Okay so here are the steps to do: 
1. I translate my files to objectiveC .h .m files.
2. I put them into my ObjectiveC project
3. I include Classes/common', 'Classes/generated/*_/_.{h,m}' from https://github.com/goodow/GDJson
4. I am done.

Is this correct? Anything missing?

Sorry for asking so much. I like your work very much. You did a great job with the lib. It helps me a lot.

**Comment by larrytin** [comment]
Right!

**Comment by drmichaelgorski** [comment]
Thank you!

**Comment by vnavarro** [comment]
@confile I checked this issue but I couldn't reproduce a good result.

We're having problems with jacksonutil.java. Do you have any samples or a guide on how to effectively  using translated source+realtime-json in iOS?

Thank you.

**Comment by drmichaelgorski** [comment]
@vnavarro Use this: http://developer.android.com/reference/org/json/package-summary.html it is supported by j2objc
